apis/ecs/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,222 @@
+package v1alpha1
+
+import xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+// CustomClusterParameters provides custom parameters for the Cluster type
+type CustomClusterParameters struct{}
+
+// CustomAWSVPCConfiguration provides custom parameters for the
+// AWSVPCConfiguration type
+type CustomAWSVPCConfiguration struct {
+	AssignPublicIP *string `json:"assignPublicIP,omitempty"`
+
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/ec2/v1beta1.SecurityGroup
+	// +crossplane:generate:reference:refFieldName=SecurityGroupRefs
+	// +crossplane:generate:reference:selectorFieldName=SecurityGroupSelector
+	SecurityGroups        []*string        `json:"securityGroups,omitempty"`
+	SecurityGroupRefs     []xpv1.Reference `json:"securityGroupRefs,omitempty"`
+	SecurityGroupSelector *xpv1.Selector   `json:"securityGroupSelector,omitempty"`
+
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/ec2/v1beta1.Subnet
+	// +crossplane:generate:reference:refFieldName=SubnetRefs
+	// +crossplane:generate:reference:selectorFieldName=SubnetSelector
+	Subnets        []*string        `json:"subnets,omitempty"`
+	SubnetRefs     []xpv1.Reference `json:"subnetRefs,omitempty"`
+	SubnetSelector *xpv1.Selector   `json:"subnetSelector,omitempty"`
+}
+
+// CustomLoadBalancer provides custom parameters for the LoadBalancer type
+type CustomLoadBalancer struct {
+	ContainerName    *string `json:"containerName,omitempty"`
+	ContainerPort    *int64  `json:"containerPort,omitempty"`
+	LoadBalancerName *string `json:"loadBalancerName,omitempty"`
```

Comment:
think we can add Ref/Selector here ? https://github.com/crossplane-contrib/provider-aws/blob/master/apis/elbv2/v1alpha1

Reply:
Sure. I'll add this.

Reply:
Done.